CONTRIBUTING.md

```diff
@@ -1,4 +1,85 @@
 # Contributing Guidelines
-1. Please maintain proper commenting etiquette. Pretend that this is a work environment and that the comments are being reviewed by coworkers.
-1. Ensure that your changes compile/work before opening a PR.
-1. Follow style guidelines where applicable.
+
+To get started with contributing to the Pitt API, please read these guidelines in their entirety to set up your development environment and learn our tooling.
+
+## Initial Setup
+
+### Setting Up Your Virtual Environment
+
+The Pitt API uses [`pipenv`](https://pipenv.pypa.io/en/latest/) to manage its dependencies. As such, you should also use `pipenv` for your own Pitt API development rather than the more traditional `pip` and `venv`.
+
+To set up `pipenv`, first install the package using `pip`:
+```sh
+pip install --user pipenv
+```
+Now install `pipenv` to your local Pitt API git repo:
+```sh
+pipenv install
+```
+This will install all necessary dependencies listed in the [Pipfile](/Pipfile) and also set up your virtual environment.
+Start the virtual envionment with
+```sh
+pipenv shell
+```
+and close it down with
+```sh
+exit
+```
+
+### Pre-Commit
+
+We highly encourage you to add [pre-commit](https://pre-commit.com/) to your development workflow in order to help us maintain a high-quality codebase.
+We've included several pre-commit hooks in [`.pre-commmit-config.yaml`](/.pre-commit-config.yaml) that we consider to be essential for our purposes.
+
+pre-commit is installed automatically as a dependency when you set up your virtual environment with `pipenv`.
+To add pre-commit to your workflow, run
+```sh
+pre-commit install
+```
+From this point on, pre-commit should run whenever you make a commit.
+If it catches any errors, it'll try to fix them automatically or ask you to fix them if it can't.
+Make sure to fix all pre-commit errors before you commit.
+
+By default, pre-commit checks all modified files staged in your commit.
+If you want to run pre-commit on the entire repo, including files that you haven't modified, run
+```sh
+pre-commit run --all-files
+```
+
+## Contributing to the API
+
+To make a contribution to the Pitt API, simply open a PR from your local fork of the Pitt API repo.
+
+Naturally, you should ensure that contributions work and are of high quality before you make a PR.
+This includes making sure that your code compiles, passes all unit tests, and adhere to common style guidelines.
+
+### Unit Testing
+
+The Pitt API uses `pytest` for unit testing, and like pre-commit it's installed automatically when you set up `pipenv`.
+To run all unit tests, run
+```sh
+pytest --cov=pittapi tests/
+```
+
+### Code Styling
+
+We don't yet require our contributors to run any code formatters as part of our pre-commit workflow or our GitHub workflows.
```

Comment:
Just require black. Keep all the code consistent.

Reply:
Fine by me, l can add it to our GitHub workflows and pre-commit hooks in another PR.

Reply:
Make another PR and link it to this one

Reply:
By the way, it looks like we're already using flake8 for linting in our GitHub Actions. How about we add that as a requirement as well?

Reply:
sgtm

Reply:
Added black and flake8 in #174, though flake8 is extremely limited for the time being based on our current configs

Reply:
@RitwikGupta Added black and flake8 to the repo in #174